lib/newProfile.ml

```diff
@@ -216,15 +243,39 @@ type settings =
 let init { output } =
   let () = assert (not (is_profiling())) in
   accu := Some { ch = output; sums = [] };
-  f "{ \"traceEvents\": [\n";
-  enter ~time:global_start "process" ();
-  enter ~time:global_start "init" ();
-  let iend = Instr.read_counter () in
-  let mend = Gc.quick_stat () in
-  let args =
-    make_instr_diff ~istart:global_start_count ~iend @
-    make_mem_diff ~mstart:global_start_stat ~mend
+  Runtime_events.start();
+
+  let () =
+    let cursor = Runtime_events.create_cursor None in
+
+    let callbacks =
+      let open Runtime_events in
+      let current_minor = ref [] in
+      let runtime_begin _ ts = function
+        | EV_MINOR -> current_minor := ts :: !current_minor
+        | _ -> ()
+      in
+      let runtime_end _ ts = function
+        | EV_MINOR -> begin match !current_minor with
```

Comment:
the events in runtime_events.mli are not documented so it's not fully clear to what extent this captures minor gc time
also not sure we can really have imbricated events so maybe we don't need the `current_minor` adhoc stack